files/build_templates/docker_image_ctl.j2

```diff
@@ -83,6 +83,13 @@ function preStartAction()
     fi
 {%- elif docker_container_name == "snmp" %}
     $SONIC_DB_CLI STATE_DB HSET 'DEVICE_METADATA|localhost' chassis_serial_number $(decode-syseeprom -s)
+{%- elif docker_container_name == "swss" %}
+    if [ ! -f /etc/sonic/asic_table.json ] && [ -f /usr/share/sonic/templates/asic_table.j2 ]; then
+        sonic-cfggen -d -t /usr/share/sonic/templates/asic_table.j2 > /etc/sonic/asic_table.json
+    fi
+    if [ ! -f /etc/sonic/peripheral_table.json ] && [ -f /usr/share/sonic/device/$PLATFORM/port_peripheral_config.j2 ]; then
+        sonic-cfggen -d -t /usr/share/sonic/device/$PLATFORM/port_peripheral_config.j2 > /etc/sonic/peripheral_table.json
+    fi
```

Comment:
Move this block into inside swss container?

Reply:
Given that the asic table and peripheral table won't be changed on-the-fly, I'd like to have them called once only when the docker is created instead of called each time the swss docker starts.
Let me know if there are other concerns.

Reply:
Seems here is called when the docker swss is started, not created.
You already check the files existing, so not too much time wasted after file created.

---
In reply to: [541481038](https://github.com/Azure/sonic-buildimage/pull/6194#discussion_r541481038) [](ancestors = 541481038)

Reply:
Looks like you are correct. This snippet of code runs during swss docker starts.
But as it works correctly and to put it here doesn't introduce any negative consequence, can we keep it as it is? To move it to swss docker needs to change the code in several places.

Reply:
ok.
Let's move to `buffermgrd.sh`. Put the code generating `asic_table.json` and `peripheral_table.json` just ahead of the place where it is used.

Reply:
Just investigated and realized it’s will be very difficult to move this part into swss docker because it requires asic_table.j2 to be copied in docker swss. However, it’s defined on a per vendor basis and in platform/<vendor> dir, which makes it difficult to copy into docker in the Dockerfile.j2.
So we have to have this part in docker_image_control.

Reply:
Regarding "it’s defined on a per vendor basis and in platform/<vendor> dir, which makes it difficult to copy into docker in the Dockerfile.j2"
I checked `docker inspect swss` and found `/usr/share/sonic/device/$PLATFORM/` is already mapped into swss container.
```
            {
                "Type": "bind",
                "Source": "/usr/share/sonic/device/x86_64-mlnx_msn2700-r0",
                "Destination": "/usr/share/sonic/platform",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
```

Mounting is happened after running a container, so I did not propose modify `Dockerfile.j2`, instead, modify `dockers/docker-orchagent/docker-init.sh`.

I think you can treat `asic_table.j2` the same way as `port_peripheral_config.j2`. Why they are copied into different folders?

---
In reply to: [541513109](https://github.com/Azure/sonic-buildimage/pull/6194#discussion_r541513109) [](ancestors = 541513109,541481038)

Reply:
The port peripheral config is defined on a per platform basis because each platform has its unique configuration. But the asic_table.j2 is shared among all platforms of the same vendor. It should not be copied to each of the platform folders.